### DISCUSSION TITLE
listen to all requests.

### DISCUSSION THREAD

**Discussion by mr-cheffy** [mitmproxy/mitmproxy#5605] [Q&A]
Hi, I am using version 5.0.0 and I would like to know how to intercept all messages from my internet connection and change their body. Currently, I can only listen to `localhost:8080` for some reason. Is there a way to listen for all URLs or should I try another software? thanks.

**Comment by Prinzhorn** [comment]
5.0.0 is almost **three years** old, I recommend always using the latest version.

See https://docs.mitmproxy.org/stable/concepts-modes/#transparent-proxy

**Reply by mr-cheffy** [reply]
yea, I tried but idk how to embed it into a python script. Because all example on how to embed it gives errors such as `mitmproxy.options.ProxyOptions don't exist` and things like that.

**Reply by Prinzhorn** [reply]
> how to embed it into a python script.

This is not a supported use-case

**Reply by Prinzhorn** [reply]
See https://github.com/mitmproxy/mitmproxy/issues/3406

**Reply by mr-cheffy** [reply]
could you recommend to me any alternatives that can be embeddable and is up to date?

**Reply by Prinzhorn** [reply]
You can embed, but on a process level (spawning a subprocess), not as a Python module. I cannot recommend anything else.

**Reply by mr-cheffy** [reply]
how could I do that? (from a CLI)

this is my current code (updated):
```python
import tempfile

tempfile_content = b"""
class ProxyProvider:

    def request(self, flow):
        if flow.response or flow.error or not flow.live:
            return
        self.run(flow)

    def response(self, flow):
        if flow.error or not flow.live:
            return
        self.run(flow)

    def run(self, flow):
        print(f"new fetch! {flow.request.url}")

addons = [ProxyProvider()]
"""

# TODO: inherit from a parent class called "Provider"
class InterfereProvider:
    options: Options

    def __init__(self, options: Options) -> None:
        self.options = options

    def execute(self):
        with tempfile.NamedTemporaryFile() as tmp:
            tmp.write(tempfile_content)
            subprocess.run(["mitmdump", "--mode", "transparent", "-s", tmp.name])
```

currently, it just stops working.

**Reply by Prinzhorn** [reply]
I cannot help you debug your own code, maybe StackOverflow is the better place for that. But yes, I am using `mitmdump` as a subprocess.
